Progress bar for downloads

Mandera opened the pull request:
After opening #388 I thought I could try doing it myself, this is the result!

### The only modified part is `WDMHttpClient.get`

- Added [tqdm](https://github.com/tqdm/tqdm) as dependency
- Changed `requests.get`'s `stream` parameter to True
  - I saw that this was the case some versions ago, hopefully, it's alright to go back to having it enabled again
- The progress bar is only shown if `Content-Length` exists in the response header and if it's greater than 100
  - Didn't want the progress bar to be shown for the "latest driver version" request which was very small
  - The value 100 should probably not be hard-coded
  - 100 might also not be the ideal default value, I only tried it on the Chrome driver
- Download speed unit is dynamic, I used the parameters from the [bytes CLI options](https://github.com/tqdm/tqdm#extra-cli-options) 
- In my attempt to mess as little as possible with the other code I'm storing the content from the stream to a bytearray
  - This bytearray then replaces the protected attribute `_content` in the response object
  - [Looks pretty safe to do](https://github.com/psf/requests/blob/da9996fe4dc63356e9467d0a5e10df3d89a8528e/requests/models.py#L889)
- I'm not too familiar with logging so not sure how to make the progress bar go through that if necessary
  - Found [this](https://stackoverflow.com/a/38739634/3936044) on SO anyhow
- Not sure how to make a unit test for this

Here's an example of how it looks
<img width="739" alt="progress" src="https://user-images.githubusercontent.com/5393746/173235384-f6bfc812-5802-41bf-bc6e-8008c6ba4656.png">

Thank you

pep8speaks commented:
Hello @Mandera! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`webdriver_manager/core/http.py`](https://github.com/SergeyPirogov/webdriver_manager/blob/cd4dfbbe60be6e487d53a86d3c91731ae9f41890/webdriver_manager/core/http.py):

> [Line 34:80](https://github.com/SergeyPirogov/webdriver_manager/blob/cd4dfbbe60be6e487d53a86d3c91731ae9f41890/webdriver_manager/core/http.py#L34): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 40:80](https://github.com/SergeyPirogov/webdriver_manager/blob/cd4dfbbe60be6e487d53a86d3c91731ae9f41890/webdriver_manager/core/http.py#L40): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 48:1](https://github.com/SergeyPirogov/webdriver_manager/blob/cd4dfbbe60be6e487d53a86d3c91731ae9f41890/webdriver_manager/core/http.py#L48): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file

SergeyPirogov commented on webdriver_manager/core/http.py:
```diff
@@ -29,6 +31,18 @@ def __init__(self):
+        resp = requests.get(url=url, verify=self._ssl_verify, stream=True, **kwargs)
         self.validate_response(resp)
+
+        total = int(resp.headers.get("Content-Length", 0))
```
can we move this to a method like show_progress or something like that?

Mandera replied:
Sure thing, I added it in utils.py though since it was static anyway

Mandera commented:
- Moved progress bar code to show_download_progress() in utils.py
- Hardcoded the "[WDM] - " prefix for the progress bar as I think making tqdm use the logger's formatting dynamically isn't feasible
- Added tqdm to install_requires in setup.py (In addition to Pipfile) - Is this the right way to do it? Seems strange to define deps in two places

SergeyPirogov commented:
Well, I think that this approach shold be just documented. I dont want to add 3rd party dependency and have a progress bar enabled by default

Mandera commented:
Yeah makes sense to have it disabled by default, add a parameter `progress_bar=False´ to `DriverManager` and its subclasses?

Are you open to making tqdm an optional dependency? Such as `pip install webdriver-manager[progress-bar]` or `pip install webdriver-manager[full]`
If not I could try writing a simplistic progress bar without 3rd party deps

SergeyPirogov commented:
> Yeah makes sense to have it disabled by default, add a parameter `progress_bar=False` to `DriverManager` and its subclasses?
> 
> Are you open to making tqdm an optional dependency? Such as `pip install webdriver-manager[progress-bar]` or `pip install webdriver-manager[full]` If not I could try writing a simplistic progress bar without 3rd party deps

Yes, making it optional is a great idea
